pageserver/src/tenant/blob_io.rs

```diff
@@ -121,10 +127,24 @@ impl<const BUFFERED: bool> BlobWriter<BUFFERED> {
     /// Writes the given buffer directly to the underlying `VirtualFile`.
     /// You need to make sure that the internal buffer is empty, otherwise
     /// data will be written in wrong order.
-    async fn write_all_unbuffered(&mut self, src_buf: &[u8]) -> Result<(), Error> {
-        self.inner.write_all(src_buf).await?;
-        self.offset += src_buf.len() as u64;
-        Ok(())
+    async fn write_all_unbuffered<B: BoundedBuf>(
```

Comment:
Could put the `#[inline(always)]` below the doc comment. Esp important if there was a huge wall of text.

Reply:
Is it worth waiting for the CI to re-run (and clicking re-run at least once more?)

Reply:
Well it appears something is broken in the CI anyway, so, will push a fix and re-run